python/ray/workflow/workflow_access.py

```diff
@@ -121,14 +123,40 @@ class LatestWorkflowOutput:
     step_id: "StepID"
 
 
+def _get_workflow_storage(workflow_id: str) -> workflow_storage.WorkflowStorage:
+    # Avoid initialize workflow in the workflow management actor,
+    # because it would invoke remote methods of itself,
+    # resulting in deadlock.
+    return workflow_storage.WorkflowStorage(workflow_id, _no_init_workflow=True)
+
+
+# TODO(suquark): what should be the default concurrency limit?
+@ray.remote(num_cpus=0)
+class WorkflowResultRespondingActor:
+    """This async actor is for responding pending workflow results."""
+
+    def __init__(self):
+        self.running_or_pending_workflows: Dict[str, asyncio.Future] = {}
+
+    def set_running_workflow_output(self, workflow_id: str, output: WorkflowStaticRef):
```

Comment:
I'm not sure, but should this function be async as well?

Reply:
Currently seem not necessary, as `await` is not used inside.